src/providers/aws.rs

```diff
@@ -428,16 +454,20 @@ impl AWSRegion {
     ) -> Result<(), Error> {
         let log = self.log.as_ref().expect("AWSRegion uninitialized");
 
+        // minimize the number of spot requests:
         for (_, reqs) in machines
             .into_iter()
             .map(|(name, m)| {
+                // attach labels (ami name, instance type)
```

Comment:
Ideally I'd also like to see this comment explain why those fields are the only ones we need to group by.